# Algerian Forest Fires Dataset Project EDA Part

## 1. UNDERSTANDING THE GIVEN TASK
#### Task Given by instructor Krish Naik
* Solve in ipynb:
1. Import the dataset.
https://archive.ics.uci.edu/ml/datasets/Algerian+Forest+Fires+Dataset++#
2. Do proper EDA(analysis) of your dataset and create a report
3. Then perform necessary preprocessing steps

## 2. DATA COLLECTION AND UNDERSTANDING

* I used a dataset on **Algerian Forest Fires** from UCI. The dataset contains a culmination of forest fire observations and data in two regions of Algeria: the Bejaia region and the Sidi Bel-Abbes region. The timeline of this dataset is from **June 2012 to September 2012**. In this project, we focused on whether certain weather features could predict forest fires in these regions using few Classification algorithms.


* As per the Task given i choose regression problem to predict fire weather index

## 3. DATA EXPLORATION

* In this step, we will apply Exploratory Data Analysis (EDA) to extract insights from the data set to know which features have contributed more in predicting Forest fire by performing Data Analysis using Pandas and Data visualization using Matplotlib & Seaborn. It is always a good practice to understand the data first and try to gather as many insights from it.


Below are tasks to be performed in EDA:

**1. Importing Libraries**


**2. Data Cleaning for EDA Report** 

**3. Exploratory Data Analysis (EDA) on all Features**

In [85]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns


# read the dataset

In [31]:
# read the data set 
df=pd.read_csv('Algerian_forest_fires_dataset_UPDATE1.csv', header=1)

In [33]:
df.head(5)

day month  year Temperature  RH  Ws Rain   FFMC  DMC    DC  ISI  BUI  FWI   
0   1     6  2012          29  57  18     0  65.7  3.4   7.6  1.3  3.4  0.5  \
1   2     6  2012          29  61  13   1.3  64.4  4.1   7.6    1  3.9  0.4   
2   3     6  2012          26  82  22  13.1  47.1  2.5   7.1  0.3  2.7  0.1   
3   4     6  2012          25  89  13   2.5  28.6  1.3   6.9    0  1.7    0   
4   5     6  2012          27  77  16     0  64.8    3  14.2  1.2  3.9  0.5   

     Classes    
0  not fire     
1  not fire     
2  not fire     
3  not fire     
4  not fire

### 3. Data Checks to perform

- Check Missing values of each columen 
- Check Duplicates
- Check data type
- Check the number of unique values of each column
- Check statistics of data set
- Check various categories present in the different categorical column

In [36]:
df.shape

(246, 14)

## Check Missing values of each columen

In [38]:
df.isnull().sum()

day            1
month          1
year           1
Temperature    1
 RH            1
 Ws            1
Rain           1
FFMC           1
DMC            1
DC             1
ISI            1
BUI            1
FWI            1
Classes        2
dtype: int64

In [48]:
df[df.isnull().any(axis=1)]

day month  year Temperature   RH   Ws Rain   FFMC   DMC      DC   ISI   
122  NaN   NaN   NaN         NaN  NaN  NaN   NaN   NaN   NaN     NaN   NaN  \
167   14     7  2012          37   37   18   0.2  88.9  12.9  14.6 9  12.5   

      BUI      FWI Classes    
122   NaN      NaN       NaN  
167  10.4  fire          NaN

In [54]:
df=df.dropna().reset_index(drop=True)

In [62]:
df.iloc[[122]]

day  month  year  Temperature   RH   Ws  Rain   FFMC  DMC  DC  ISI  BUI   
122  day  month  year  Temperature   RH   Ws  Rain   FFMC  DMC  DC  ISI  BUI  \

     FWI  Classes    
122  FWI  Classes

In [63]:
df=df.drop(122).reset_index(drop=True)

In [64]:
df.iloc[[122]]

day month  year Temperature  RH  Ws Rain   FFMC  DMC   DC  ISI  BUI  FWI   
122   1     6  2012          32  71  12   0.7  57.1  2.5  8.2  0.6  2.8  0.2  \

       Classes    
122  not fire

#### Insights or Observation

- conclussion:-- in row no. 122  all vluse is null and in row no. 167 last column value is null  is pasent in data set

1. drop null value rows
2. drop row 122 which contain heading again

In [65]:
df.isna().sum()

day            0
month          0
year           0
Temperature    0
 RH            0
 Ws            0
Rain           0
FFMC           0
DMC            0
DC             0
ISI            0
BUI            0
FWI            0
Classes        0
dtype: int64

Insights or Observation
conclussion:-- in each columen 1 missing (in classes==2) vluse is pasent

## Check Duplicates

In [42]:
df.duplicated().sum()

0

#### Insights or Observation

- conclussion:-- no duplicate valuse is present

## Check the number of unique values of each column


In [44]:
df.nunique()

day             32
month            5
year             2
Temperature     20
 RH             63
 Ws             19
Rain            40
FFMC           174
DMC            167
DC             199
ISI            107
BUI            175
FWI            127
Classes          8
dtype: int64

#### Insights or Observation

- conclussion:-- no duplicate valuse is present

## Check various categories present in the different categorical column

In [66]:
df.columns

Index(['day', 'month', 'year', 'Temperature', ' RH', ' Ws', 'Rain ', 'FFMC',
       'DMC', 'DC', 'ISI', 'BUI', 'FWI', 'Classes  '],
      dtype='object')

### to remove space between heading name

In [68]:
df.columns=df.columns.str.strip()

In [69]:
df.columns

Index(['day', 'month', 'year', 'Temperature', 'RH', 'Ws', 'Rain', 'FFMC',
       'DMC', 'DC', 'ISI', 'BUI', 'FWI', 'Classes'],
      dtype='object')

In [71]:
df.head(10)

day month  year Temperature  RH  Ws  Rain  FFMC   DMC    DC  ISI   BUI  FWI   
0   1     6  2012          29  57  18     0  65.7   3.4   7.6  1.3   3.4  0.5  \
1   2     6  2012          29  61  13   1.3  64.4   4.1   7.6    1   3.9  0.4   
2   3     6  2012          26  82  22  13.1  47.1   2.5   7.1  0.3   2.7  0.1   
3   4     6  2012          25  89  13   2.5  28.6   1.3   6.9    0   1.7    0   
4   5     6  2012          27  77  16     0  64.8     3  14.2  1.2   3.9  0.5   
5   6     6  2012          31  67  14     0  82.6   5.8  22.2  3.1     7  2.5   
6   7     6  2012          33  54  13     0  88.2   9.9  30.5  6.4  10.9  7.2   
7   8     6  2012          30  73  15     0  86.6  12.1  38.3  5.6  13.5  7.1   
8   9     6  2012          25  88  13   0.2  52.9   7.9  38.8  0.4  10.5  0.3   
9  10     6  2012          28  79  12     0  73.2   9.5  46.3  1.3  12.6  0.9   

       Classes  
0  not fire     
1  not fire     
2  not fire     
3  not fire     
4  not fire     
5      fire     
6      fire     
7      fire     
8  not fire     
9  not fire

## Check data type

In [72]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 243 entries, 0 to 242
Data columns (total 14 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   day          243 non-null    object
 1   month        243 non-null    object
 2   year         243 non-null    object
 3   Temperature  243 non-null    object
 4   RH           243 non-null    object
 5   Ws           243 non-null    object
 6   Rain         243 non-null    object
 7   FFMC         243 non-null    object
 8   DMC          243 non-null    object
 9   DC           243 non-null    object
 10  ISI          243 non-null    object
 11  BUI          243 non-null    object
 12  FWI          243 non-null    object
 13  Classes      243 non-null    object
dtypes: object(14)
memory usage: 26.7+ KB


#### Insights or Observation

- conclussion:-- 
1. day, month, year, Temperature, RH, Ws === is integer data (but info gives it object which is wrong)
2. Rain,FFMC,DMC,DC,ISI,BUI,FWI == is flote data (but info gives it object which is wrong)

In [78]:
df[['day','month','year','Temperature','RH','Ws']]=df[['day','month','year','Temperature','RH','Ws']].astype(int)

In [79]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 243 entries, 0 to 242
Data columns (total 14 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   day          243 non-null    int32 
 1   month        243 non-null    int32 
 2   year         243 non-null    int32 
 3   Temperature  243 non-null    int32 
 4   RH           243 non-null    int32 
 5   Ws           243 non-null    int32 
 6   Rain         243 non-null    object
 7   FFMC         243 non-null    object
 8   DMC          243 non-null    object
 9   DC           243 non-null    object
 10  ISI          243 non-null    object
 11  BUI          243 non-null    object
 12  FWI          243 non-null    object
 13  Classes      243 non-null    object
dtypes: int32(6), object(8)
memory usage: 21.0+ KB


In [80]:
df[['Rain','FFMC',	'DMC',	'DC',	'ISI',	'BUI',	'FWI']]=df[['Rain','FFMC',	'DMC',	'DC',	'ISI',	'BUI',	'FWI']].astype(float)

In [81]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 243 entries, 0 to 242
Data columns (total 14 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   day          243 non-null    int32  
 1   month        243 non-null    int32  
 2   year         243 non-null    int32  
 3   Temperature  243 non-null    int32  
 4   RH           243 non-null    int32  
 5   Ws           243 non-null    int32  
 6   Rain         243 non-null    float64
 7   FFMC         243 non-null    float64
 8   DMC          243 non-null    float64
 9   DC           243 non-null    float64
 10  ISI          243 non-null    float64
 11  BUI          243 non-null    float64
 12  FWI          243 non-null    float64
 13  Classes      243 non-null    object 
dtypes: float64(7), int32(6), object(1)
memory usage: 21.0+ KB


## Check statistics of data set

In [84]:
df.describe().T

count         mean        std     min      25%     50%      75%   
day          243.0    15.761317   8.842552     1.0     8.00    16.0    23.00  \
month        243.0     7.502058   1.114793     6.0     7.00     8.0     8.00   
year         243.0  2012.000000   0.000000  2012.0  2012.00  2012.0  2012.00   
Temperature  243.0    32.152263   3.628039    22.0    30.00    32.0    35.00   
RH           243.0    62.041152  14.828160    21.0    52.50    63.0    73.50   
Ws           243.0    15.493827   2.811385     6.0    14.00    15.0    17.00   
Rain         243.0     0.762963   2.003207     0.0     0.00     0.0     0.50   
FFMC         243.0    77.842387  14.349641    28.6    71.85    83.3    88.30   
DMC          243.0    14.680658  12.393040     0.7     5.80    11.3    20.80   
DC           243.0    49.430864  47.665606     6.9    12.35    33.1    69.10   
ISI          243.0     4.742387   4.154234     0.0     1.40     3.5     7.25   
BUI          243.0    16.690535  14.228421     1.1     6.00    12.4    22.65   
FWI          243.0     7.035391   7.440568     0.0     0.70     4.2    11.45   

                max  
day            31.0  
month           9.0  
year         2012.0  
Temperature    42.0  
RH             90.0  
Ws             29.0  
Rain           16.8  
FFMC           96.0  
DMC            65.9  
DC            220.4  
ISI            19.0  
BUI            68.0  
FWI            31.1

# counts of unique values.

In [88]:
df.Classes.value_counts()

Classes
fire             131
not fire         100
fire               4
not fire           3
fire               2
not fire           2
not fire           1
Name: count, dtype: int64

#### Looks like the values were misspaced so we can use **str.strip()** method

In [91]:
df.Classes=df.Classes.str.strip()
df.Classes.value_counts()

Classes
fire        137
not fire    106
Name: count, dtype: int64

## Region 1 (Bejaia Region)

In [92]:
df[:122]

day  month  year  Temperature  RH  Ws  Rain  FFMC  DMC    DC  ISI  BUI   
0      1      6  2012           29  57  18   0.0  65.7  3.4   7.6  1.3  3.4  \
1      2      6  2012           29  61  13   1.3  64.4  4.1   7.6  1.0  3.9   
2      3      6  2012           26  82  22  13.1  47.1  2.5   7.1  0.3  2.7   
3      4      6  2012           25  89  13   2.5  28.6  1.3   6.9  0.0  1.7   
4      5      6  2012           27  77  16   0.0  64.8  3.0  14.2  1.2  3.9   
..   ...    ...   ...          ...  ..  ..   ...   ...  ...   ...  ...  ...   
117   26      9  2012           31  54  11   0.0  82.0  6.0  16.3  2.5  6.2   
118   27      9  2012           31  66  11   0.0  85.7  8.3  24.9  4.0  9.0   
119   28      9  2012           32  47  14   0.7  77.5  7.1   8.8  1.8  6.8   
120   29      9  2012           26  80  16   1.8  47.4  2.9   7.7  0.3  3.0   
121   30      9  2012           25  78  14   1.4  45.0  1.9   7.5  0.2  2.4   

     FWI   Classes  
0    0.5  not fire  
1    0.4  not fire  
2    0.1  not fire  
3    0.0  not fire  
4    0.5  not fire  
..   ...       ...  
117  1.7  not fire  
118  4.1      fire  
119  0.9  not fire  
120  0.1  not fire  
121  0.1  not fire  

[122 rows x 14 columns]

## Region 2 (Sidi-Bel Abbes Region)

In [94]:
df[122:]

day  month  year  Temperature  RH  Ws  Rain  FFMC   DMC    DC  ISI   BUI   
122    1      6  2012           32  71  12   0.7  57.1   2.5   8.2  0.6   2.8  \
123    2      6  2012           30  73  13   4.0  55.7   2.7   7.8  0.6   2.9   
124    3      6  2012           29  80  14   2.0  48.7   2.2   7.6  0.3   2.6   
125    4      6  2012           30  64  14   0.0  79.4   5.2  15.4  2.2   5.6   
126    5      6  2012           32  60  14   0.2  77.1   6.0  17.6  1.8   6.5   
..   ...    ...   ...          ...  ..  ..   ...   ...   ...   ...  ...   ...   
238   26      9  2012           30  65  14   0.0  85.4  16.0  44.5  4.5  16.9   
239   27      9  2012           28  87  15   4.4  41.1   6.5   8.0  0.1   6.2   
240   28      9  2012           27  87  29   0.5  45.9   3.5   7.9  0.4   3.4   
241   29      9  2012           24  54  18   0.1  79.7   4.3  15.2  1.7   5.1   
242   30      9  2012           24  64  15   0.2  67.3   3.8  16.5  1.2   4.8   

     FWI   Classes  
122  0.2  not fire  
123  0.2  not fire  
124  0.1  not fire  
125  1.0  not fire  
126  0.9  not fire  
..   ...       ...  
238  6.5      fire  
239  0.0  not fire  
240  0.2  not fire  
241  0.7  not fire  
242  0.5  not fire  

[121 rows x 14 columns]

# load the cleaned data in new file 

In [95]:
df.to_csv('Algerian_forest_fires_dataset_CLEANED.csv', index=False)

# 3.5 Exploratory Data Analysis (EDA)